Start using prettier

dbrgn opened the pull request:
Updated version of #32.

To check formatting:

    npx prettier --check .

To reformat:

    npx prettier -w .

dbrgn committed:
Set up prettier

Automatic code formatting with prettier gives us consistent code style
without the need for any stylistic discussions.

When using the recommended VSCode extension, code is automatically
formatted on save.

Adherence to the prettier code style is checked in CI.

dbrgn committed:
Prevent git from converting line endings to CRLF on Windows

noppej commented on src/extension.ts:
```diff
@@ -7,372 +7,452 @@
+
+            var options = {
+                cwd: session.configuration.cwd,
+                // eslint-disable-next-line @typescript-eslint/naming-convention
```
With the new formatting, this eslint hint no longer covers RUST_LOG and DEFMT_LOG, and results in warnings. Can you please move the hint so that the warnings go away.

dbrgn replied:
This is now obsolete, since the eslint comment is gone.

noppej commented on README.md:
```diff
@@ -1,65 +1,91 @@
+To work on this extensions, you first need to install VS Code and nodejs.
+Afterwards, follow the following steps:
+
+* Install yarn:
```
By removing the ``` modifiers from lines like this, it removes the affordances in webview that allows users to easily copy these as lines of code. Is there a particular reason why you removed them all?

dbrgn replied:
Uhh, that's a mistake on my side, I'll try to fix this. (Seems like just 4 chars of indentation is not sufficient when part of a list.)

noppej replied:
@dbrgn Can you please add the ``` back, so that users an get the clipboard functions to copy these lines for re-use?

dbrgn replied:
~~Ah, now I get what you mean. I personally don't find that important at all (you can just double-click the text, then copy), and it makes the non-rendered version of the markdown file harder to read, but I'll revert this change if you prefer the triple-backtick version!~~

dbrgn replied:
For a short moment I thought that the "copy" icon only appears with the triple-backtick version, but that doesn't seem to be the case.

Here's the rendered version: https://github.com/probe-rs/vscode/tree/prettier The copy icon appears at the right when hovering the element.

![image](https://user-images.githubusercontent.com/105168/210169408-2e5b815e-246a-44c6-823e-0c2bc8a0c5e9.png)

But maybe I'm misunderstanding you 🙂

noppej replied:
> I thought that the "copy" icon only appears with the triple-backtick version, but that doesn't seem to be the case.

@dbrgn, it depends. In web view, it shows up, but in VSCode Preview of MD it doesn't show up.  I'll accept this as is, even though I don't understand why prettier insists on taking away the triple-backtick.

noppej reviewed (changes requested):
Thank you for these changes. I have made two minor comments/requests, but other than that, this is a nice improvement to the repo. 

You mentioned at some point that you might want to remove the yarn dependency. Did you change your mind on that?

dbrgn commented:
> You mentioned at some point that you might want to remove the yarn dependency. Did you change your mind on that?

No, but I'll do that in a separate PR 🙂

noppej commented:
@dbrgn Just putting this back on your radar ... I'd hate to loose this contribution from you :)

dbrgn committed:
Reformat code with prettier

Done using `npx prettier -w .`

dbrgn committed:
Add .git-blame-ignore-revs file

To use it, invoke `git blame` like this:

    git blame --ignore-revs-file .git-blame-ignore-revs

Alternatively, make this the default:

    git config [--global] blame.ignoreRevsFile .git-blame-ignore-revs

dbrgn committed:
Manually format README for better readability

- Consistent lists
- Code blocks with indentation
- Line wrapping at 80 characters

dbrgn commented:
@noppej finally updated, just before the end of the year 😂

For commit [a6cce30](https://github.com/probe-rs/vscode/pull/34/commits/a6cce305d79630bb56cc70f6e0b41c3a1a170137) not to break, this MR may not be rebased. (Otherwise, the commit reference must be updated.)

noppej reviewed (changes requested):
Thanks for getting back to this ... I don't have experience with this part of it, so really appreciate your help. 

Only have one small request left over in https://github.com/probe-rs/vscode/pull/34#discussion_r1059664725

Yatekii reviewed (approved):
Thank you :)

bors r+

Yatekii commented:
bors retry

Yatekii commented:
Ah lol, I didn't know this does not work :D

noppej reviewed (approved):
Thanks again for this. I will not bump the release version on this, since the changes are not user facing.

bors r+
